torchx/schedulers/__init__.py

```diff
@@ -22,6 +22,17 @@ def __call__(self, session_name: str, **kwargs: object) -> Scheduler:
         ...
 
 
+def add_optional_ray_scheduler_factory(factories: Dict[str, SchedulerFactory]) -> None:
```

Comment:
we generally handle missing imports by importing within the relevant functions. Makes it a bit simpler usage wise

i.e.
https://github.com/pytorch/torchx/blob/main/torchx/schedulers/kubernetes_scheduler.py#L361

for top level types you can throw them behind TYPE_CHECKING:
https://github.com/pytorch/torchx/blob/main/torchx/schedulers/kubernetes_scheduler.py#L64

Reply:
Do you mean within `ray_scheduler.create_scheduler()`?

Reply:
Ok, I think I get what you mean after checking the docker scheduler. I revised the implemented, please check the latest revision.